allensdk/test/internal/brain_observatory/test_time_sync.py

```diff
@@ -544,3 +545,127 @@ def mock_read_pickle(*args, **kwargs):
+         "behavior_camera": "cam1_exposure",
+         "acquiring": "2p_acquiring",
+         "lick_sensor": "lick_1"
+     }),
+    (None, ['2p_vsync', 'stim_vsync', 'stim_photodiode', 'acq_trigger',
+            '', 'cam1_exposure', 'cam2_exposure'],
+     {
+         "photodiode": "stim_photodiode",
+         "2p": "2p_vsync",
+         "stimulus": "stim_vsync",
+         "eye_camera": "cam2_exposure",
+         "behavior_camera": "cam1_exposure",
+         "acquiring": "2p_acquiring",
+         "lick_sensor": "lick_1"
+     }),
+    (None, [],
+     {
+         "photodiode": "stim_photodiode",
+         "2p": "2p_vsync",
+         "stimulus": "stim_vsync",
+         "eye_camera": "cam2_exposure",
+         "behavior_camera": "cam1_exposure",
+         "acquiring": "2p_acquiring",
+         "lick_sensor": "lick_1"
+     }),
+    (None, ['', 'stim_vsync', 'photodiode', 'acq_trigger', 'eye_tracking',
+            'lick_1'],
+     {
+         "photodiode": "photodiode",
+         "2p": "2p_vsync",
+         "stimulus": "stim_vsync",
+         "eye_camera": "eye_tracking",
+         "behavior_camera": "cam1_exposure",
+         "acquiring": "2p_acquiring",
+         "lick_sensor": "lick_1"
+     }),
+    (None, ['', '2p_vsync', 'photodiode', 'acq_trigger', 'eye_tracking',
+            'lick_1'],
+     {
+         "photodiode": "photodiode",
+         "2p": "2p_vsync",
+         "stimulus": "stim_vsync",
+         "eye_camera": "eye_tracking",
+         "behavior_camera": "cam1_exposure",
+         "acquiring": "2p_acquiring",
+         "lick_sensor": "lick_1"
+     }),
+    (None, ['barcode_ephys', 'vsync_stim', 'stim_photodiode', 'stim_running',
+            'beh_frame_received', 'eye_frame_received', 'face_frame_received',
+            'stim_running_opto', 'stim_trial_opto', 'face_came_frame_readout',
+            'eye_cam_frame_readout', 'beh_cam_frame_readout',
+            'face_cam_exposing', 'eye_cam_exposing', 'beh_cam_exposing',
+            'lick_sensor'],
+     {
+         "photodiode": "stim_photodiode",
+         "2p": "2p_vsync",
+         "stimulus": "vsync_stim",
+         "eye_camera": "eye_frame_received",
+         "behavior_camera": "beh_frame_received",
+         "acquiring": "2p_acquiring",
+         "lick_sensor": "lick_sensor"
+     })
+])
+def test_get_keys(sync_dset, line_labels, expected_line_labels):
+    """
+    Test Cases:
+        1) Test Case from Wayne running a job on this branch,
+           simple case with all keys present
+        2) Test Case with alternate key set
+        3) Test Case with eye camera key missing
+        4) Test Case with behavior camera key missing
+        5) Test Case with lick sensor key missing
+        6) Test Case with all keys missing
+        7) Test Case with 2p key missing
+        8) Test Case with stimulus key missing
+        9) Test Case with V3 keys
+
+    """
+    ds = MockSyncDataset(None, line_labels)
+    keys = ts.get_keys(ds)
+    assert keys == expected_line_labels
```

Comment:
Please add coverage for your warning message.
In addition, consider edge cases and functional distinctions in code when writing unit tests. It's unwieldy and in this case unnecessary to cover all the individual possible missing fields, since some of them are functionally the same.

The cases I'd recommend are:
Missing one key (of type that has only one option, since you differentiate between lists and strings in your key function)
Missing one key (that has multiple options, list type)
Missing more than one key
All keys present

Reply:
Done

Reply:
I don't think these are labeled correctly.

Reply:
Also it looks like your first `expected` test case is wrong. I don't see the "2p_acquiring" label in the input list so it shouldn't be in the expected output. Are you creating the expected result before running the function or after?

Reply:
Please also do not resolve changes requested by others unless they are very simple (e.g. a typo). I'll resolve it once see that the request has been completed.

Reply:
2p_acquiring is not in the line labels for the test case failing on bamboo but is required at line 276 of time_sync.py. It's needed to get the ophys_timestamps so I kept it as defaulted to 2p_acquiring as it is not in the line labels for the ophys experiments on bamboo.

Reply:
The other keys are also required in some functions, because they are data sources and this is a data transformer. It doesn't make sense to provide a default for "acquiring" source and not any others.

Your function also doesn't check whether "2p_acquiring" is actually present in the data, but checks the others and provides a warning message if they are missing. This inconsistency would confuse a user, since there's no reason for this line to behave differently than others.

Reply:
Fixed this by changing 2p_acquiring to non default and working with the same behavior as the other potential key value pairs.

Reply:
@isaak-willett I can't approve this PR until @kschelonka  resolves this conversation.

Reply:
I like the update so that everything is just in a list in the prospective schema. Once the unnecessary conditional is removed I will approve.